### PR TITLE
Implement view screenshot feature

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: eq/fabric/view.h [hunk 1]
```diff
@@ -208,6 +208,18 @@ class View : public Object, public Frustum
      * @version 1.0
      */
     EQFABRIC_INL uint64_t getCapabilities() const;
+
+    /**
+     * Enable recording of given buffers for screenshot feature.
+     * @version 2.1
```
Comment (eile) [inline eq/fabric/view.h]: Make Frame::Buffer an enum class and use here (as separate PR?)

File: examples/eqPly/config.cpp [hunk 2]
```diff
@@ -233,7 +233,23 @@ uint32_t Config::startFrame()
...
+        const auto& image = view->pollScreenshot();
+        if( image.hasPixelData( eq::Frame::BUFFER_COLOR ))
+        {
+            const std::string screenshotPath = "./eqPly_" + std::to_string( frameNumber ) + ".png";
+            image.writeImage( screenshotPath, eq::Frame::BUFFER_COLOR );
+            std::ostringstream os;
+            os << "Screenshot written to " <<  screenshotPath;
+            _setMessage( os.str( ));
+        }
+    }
```
Comment (eile) [inline examples/eqPly/config.cpp]: refactor into method

File: examples/eqPly/config.cpp [hunk 3]
```diff
@@ -233,7 +233,23 @@ uint32_t Config::startFrame()
...
+    const auto frameNumber = eq::Config::startFrame( version );
+
+    View* view = _getCurrentView();
+    if( view )
+    {
+        view->setRecording( eq::Frame::BUFFER_NONE );
+        const auto& image = view->pollScreenshot();
+        if( image.hasPixelData( eq::Frame::BUFFER_COLOR ))
+        {
+            const std::string screenshotPath = "./eqPly_" + std::to_string( frameNumber ) + ".png";
```
Comment (eile) [inline examples/eqPly/config.cpp]: use image->frameNumber

File: doc/Changelog.md [hunk 4]
```diff
@@ -3,6 +3,9 @@ Changelog {#Changelog}
 
 # git master
 
+* [607](https://github.com/Eyescale/Equalizer/pull/607):
+  Implement view screenshot feature: eq::View::setRecording() and
+  eq::View::pollScreenshot()
```
Comment (eile) [inline doc/Changelog.md]: void enableScreenshot( buffers, func( frameNumber, image ));
    void disableScreenshot();

File: eq/fabric/view.h (hunk 1 above)
Comment (tribal-tec) [inline eq/fabric/view.h]: Done

File: examples/eqPly/config.cpp (hunk 3 above)
Comment (tribal-tec) [inline examples/eqPly/config.cpp]: Done

File: examples/eqPly/config.cpp (hunk 2 above)
Comment (tribal-tec) [inline examples/eqPly/config.cpp]: Done. Lambda now :)

File: doc/Changelog.md (hunk 4 above)
Comment (tribal-tec) [inline doc/Changelog.md]: Done

File: eq/view.cpp [hunk 5]
```diff
@@ -161,6 +201,53 @@ bool View::handleEvent( const EventType type, const SizeEvent& event )
...
+    const auto& newImage = command.read< Image >();
+
+    auto& screenshot = _impl->screenshot[ frameNumber ];
+    screenshot.composite( vp, newImage );
+
+    if( screenshot.isComplete( ))
+    {
+        _impl->screenshotFunc( frameNumber, *screenshot.image );
+        _impl->screenshot.erase( frameNumber );
+        _impl->screenshotFunc = ScreenshotFunc();
```
Comment (eile) [inline eq/view.cpp]: don't remove

**Pull request merged**